File: api/src/main/java/de/bytemc/tournaments/api/BooleanResult.java
package de.bytemc.tournaments.api;

/**
 * @author Nico_ND1
 */
public final class BooleanResult {

    public static final BooleanResult TRUE = new BooleanResult(true);
    public static final BooleanResult FALSE = new BooleanResult(false);

    public static BooleanResult from(boolean result) {
        return new BooleanResult(result);
    }

    private final boolean result;

    public BooleanResult(boolean result) {
        this.result = result;
    }

    public boolean getResult() {
        return result;
    }
}

File: api/src/main/kotlin/de/bytemc/tournaments/api/BooleanResult.kt
package de.bytemc.tournaments.api

/**
 * @author Nico_ND1
 */
data class BooleanResult constructor(val result: Boolean) {
    companion object {
        val TRUE = BooleanResult(true)
        val FALSE = BooleanResult(false)

        fun from(boolean: Boolean): BooleanResult {
            return if (boolean) TRUE else FALSE
        }
    }
}
